add new kb about semgrep login and single-tenant customers (#1579)
semgrep/semgrep-docs: docs/kb/semgrep-appsec-platform/semgrep-login-cli-tenant.md

---

> description: Fix a semgrep login issue for single-tenant customers.

Reviewer: ```suggestion
description: Execute `semgrep login` correctly for customers on dedicated tenants.
```

tags:
  - Semgrep AppSec Platform
  - Login
  - Troubleshooting
---

> ## semgrep login doesn't redirect to my Semgrep site.

Reviewer: ```suggestion
## The `semgrep login` command doesn't redirect to my Semgrep tenant site
```

When executing the command: 
```console
semgrep login
```

> it redirects to the default site: `https://semgrep.dev` and it could lead to an authentication issue if you are a single-tenant customers, as the expected site should be an URL like: `https://yourcompany.semgrep.dev`

Reviewer: This line seems unclear to me. It definitely will lead to an issue, and we should be specific about what the issue is (unable to log in/activate token). I think I'd write something like this:

it redirects to [https://semgrep.dev](https://semgrep.dev), and I receive an error {whatever the error is}. I'm expecting to be directed to `https://mycompany.semgrep.dev`.

> ## How to fix this login issue

Reviewer: ```suggestion
## To log in to the correct tenant site
```

> It can be solved setting the environment variable `SEMGREP_APP_URL` before calling to the `semgrep login` function.

Reviewer 2: ```suggestion
It can be solved by setting the environment variable `SEMGREP_APP_URL` before calling the `semgrep login` function.
```

> It can be solved by setting the environment variable `SEMGREP_APP_URL` before calling the `semgrep login` function.

Reviewer: ```suggestion
Set the environment variable `SEMGREP_APP_URL` before calling the `semgrep login` function.
```

```console

> export SEMGREP_APP_URL=https://yourcompany.semgrep.dev

Reviewer: ```suggestion
export SEMGREP_APP_URL=https://mycompany.semgrep.dev
semgrep login
```

```

> As login can be a recurrent task, one tip is to set the export statement in the `.~/.bashrc` file or similar depending on your operative system.

Reviewer: ```suggestion
If you frequently log in from the command line, set the SEMGREP_APP_URL variable in your shell initialization file, such as `~/.zshrc` or `~ /.bash_profile`, depending on your operating system.
```

---
description: Execute `semgrep login` correctly for customers on dedicated tenants.
tags:
  - Semgrep AppSec Platform
  - Login
  - Troubleshooting
---

> ## The `semgrep login` command doesn't redirect to my Semgrep tenant site

Reviewer 2: ```suggestion
# The `semgrep login` command doesn't redirect to my Semgrep tenant site
```
Should this be the title?

When executing the command: 
```console
semgrep login
```

> it redirects to `https://semgrep.dev`, and I receive an error when login: `The requested URL was not found on the server`. I'm expecting to be redirected to `https://mycompany.semgrep.dev`

Reviewer 2: ```suggestion
it redirects to `https://semgrep.dev`. You may receive an error when logging in: `The requested URL was not found on the server`. As a Semgrep tenant user,  you should be redirected to your tenant site, such as: <code>https://<span className="placeholder">MY_COMPANY</span>.semgrep.dev</code>
```

Reviewer: This is switching person in the same para now - maybe we should make the "You may..." a new para? Or change the title, I know the use of "my" is somewhat contentious.
